File: src-tauri/src/commands/homepage.rs
use crate::error::CommandError;
use crate::models::homepage::{HomepageData, NewsResponse};
use crate::services::homepage_service;
use crate::state::AppState;
use tauri::State;

/// Get aggregated homepage data (screenshots, instances, worlds)
#[tauri::command]
pub fn get_homepage_data(state: State<'_, AppState>) -> Result<HomepageData, CommandError> {
    homepage_service::get_homepage_data(&state).map_err(CommandError::from)
}

/// Get Minecraft news articles
#[tauri::command]
pub async fn get_minecraft_news(state: State<'_, AppState>) -> Result<NewsResponse, CommandError> {
    homepage_service::fetch_minecraft_news(&state.http_client)
        .await
        .map_err(CommandError::from)
}

File: src/lib/services/homepage.ts
import { invoke } from '@tauri-apps/api/core';
import type { HomepageData, NewsResponse } from '$lib/types';

/**
 * Get aggregated homepage data (screenshots, instances, worlds)
 */
export async function getHomepageData(): Promise<HomepageData> {
	return invoke<HomepageData>('get_homepage_data');
}

/**
 * Get Minecraft news articles
 */
export async function getMinecraftNews(): Promise<NewsResponse> {
	return invoke<NewsResponse>('get_minecraft_news');
}
